File: 0645-set-mismatch/0645-set-mismatch.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        s=set([i for i in range(1, len(nums)+1)])
        ans=[-1, -1]
        for i in range(len(nums)):
            if nums[i] in s:
                s.remove(nums[i])
            else:
                ans[0]=nums[i]
        ans[1]=s.pop()
        return ans

File: 0645-set-mismatch/0645-set-mismatch.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        int n = nums.length;
        Set<Integer> set = new HashSet<>();
        boolean[] map = new boolean[n+1];
        int repeat = 0;
        for(int i: nums){
            if(set.contains(i)){
                repeat = i;
            }
            set.add(i);
            map[i]=true;
        }
        for(int i=1; i<n+1; ++i){
            if(!map[i]) return new int[]{repeat, i};
        }
        
        return new int[]{0, 0};
    }
}